crates/wasmtime/src/runtime/vm/table.rs

```diff
@@ -349,7 +354,11 @@ impl Table {
     }
 
     fn limit_new(plan: &TablePlan, store: &mut dyn Store) -> Result<()> {
-        if !store.table_growing(0, plan.table.minimum, plan.table.maximum)? {
+        if !store.table_growing(
+            0,
+            plan.table.minimum.try_into().unwrap(),
+            plan.table.maximum.map(|i| i.try_into().unwrap()),
+        )? {
```

Comment:
Here I might recommend duplicating (in spirit at least) what's happening in `vm/memory.rs`. Namely the `table_growing `callback` is always called no matter what but the min/max passed in are `usize::MAX` if they overflow. Notably this shouldn't `.unwrap()` here but you'll want to try to convert to a `usize`.

After the `table_growing` call is made though you'll convert the min/max into `usize` and return an error if something overflows.

Also like `Memory::limit_new` I'd recomend returning `(usize, Option<usize>)` from this method to avoid re-casting things in the callers and reusing the results of this function.